### ISSUE TITLE
SST build issue on Titan

### ISSUE THREAD

**Issue by kshitij-v-mehta** [ornladios/ADIOS2#533]
Building on Titan fails with the following error. Built as 
```
cmake -DCMAKE_INSTALL_PREFIX=$HOME/opt/adios2/installation/ -DADIOS2_USE_DataMan=OFF -DADIOS2_USE_HDF5=OFF -DADIOS2_USE_Fortran=ON -DCMAKE_BUILD_TYPE=Release ../
```

```
[ 38%] Built target sst
Scanning dependencies of target adios2
../../../lib/libadios2_enet.a(unix.c.o): In function `enet_address_set_host':
/ccs/home/kmehta/opt/adios2/thirdparty/enet/enet/unix.c:126: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
../../../lib/libadios2_evpath.a(ip_config.c.o): In function `get_qual_hostname':
/ccs/home/kmehta/opt/adios2/thirdparty/EVPath/EVPath/ip_config.c:344: warning: Using 'gethostbyaddr' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
../../../lib/libadios2_evpath.a(ip_config.c.o): In function `get_self_ip_iface':
/ccs/home/kmehta/opt/adios2/thirdparty/EVPath/EVPath/ip_config.c:110: warning: Using 'gethostbyname' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
../../../lib/libadios2_ffs.a(standard.c.o): In function `dlload_externs':
/ccs/home/kmehta/opt/adios2/thirdparty/ffs/ffs/cod/standard.c:687: undefined reference to `dlopen'
/ccs/home/kmehta/opt/adios2/thirdparty/ffs/ffs/cod/standard.c:690: undefined reference to `dlsym'
/usr/bin/ld: link errors found, deleting executable `../../../bin/cmprobe'
/usr/bin/sha1sum: ../../../bin/cmprobe: No such file or directory
collect2: error: ld returned 1 exit status
make[2]: *** [bin/cmprobe] Error 1
```

Missing `-ldl` somewhere?

**Comment by eisenhauer** [comment]
Hi Kshitij.  For Titan, you probably have to tell us what compiler, etc.  Just giving us a list of what modules are loaded when you get this would be good...

**Comment by kshitij-v-mehta** [comment]
I am using gcc/4.9.3. The loaded modules are 
```
Currently Loaded Modulefiles:
  1) eswrap/1.3.3-1.020200.1280.0
  2) craype-network-gemini
  3) craype/2.5.13
  4) cray-mpich/7.6.3
  5) craype-interlagos
  6) lustredu/1.4
  7) xalt/0.7.5
  8) git/2.13.0
  9) module_msg/0.1
 10) modulator/1.2.0
 11) hsi/5.0.2.p1
 12) DefApps
 13) gcc/4.9.3
 14) cray-libsci/16.11.1
 15) udreg/2.3.2-1.0502.10518.2.17.gem
 16) ugni/6.0-1.0502.10863.8.28.gem
 17) pmi/5.0.12
 18) dmapp/7.0.1-1.0502.11080.8.74.gem
 19) gni-headers/4.0-1.0502.10859.7.8.gem
 20) xpmem/0.1-2.0502.64982.5.3.gem
 21) dvs/2.5_0.9.0-1.0502.2188.1.113.gem
 22) alps/5.2.4-2.0502.9774.31.12.gem
 23) rca/1.0.0-2.0502.60530.1.63.gem
 24) atp/2.1.1
 25) PrgEnv-gnu/5.2.82
 26) python/3.5.1
 27) cudatoolkit/7.5.18-1.0502.10743.2.1
 28) cmake3/3.6.1
```

**Comment by kshitij-v-mehta** [comment]
Also, I am building current adios2 head.

**Comment by chuckatkins** [comment]
I was able to easily reproduce the issue.  See upstream fix in GTkorvo/ffs#6.  Once merged I'll pull it back down into ADIOS2

**Comment by chuckatkins** [comment]
Cant close until upstream propagates
